### DISCUSSION TITLE
Why does the NextJS example use NEXT_PUBLIC_ for environment variables?

### DISCUSSION THREAD

**Discussion by aminamos** [supabase/supabase#6877] [Questions]
Hi,

Example: nextjs-with-supabase-auth

This repo uses the public environment variable option to expose the supabase-js client to the browser. Some would recommend against using this option because it makes the variable values visible in the build. I vaguely remember reading somewhere on the Supabase site that this is fine as long as row level security is enabled, but I’m not sure how RLS mitigates exposing keys publicly. 

Is there an example that uses private environment variables? From what I understand I’d have to change the flow a bit and only use the Supabase client on the backend with this option. 

Thanks!

**Comment by zvitez** [comment]
I had the same question. Based on what I understood from [the docs](https://supabase.com/docs/learn/auth-deep-dive/auth-row-level-security), you can't really do much with the anon_key if RLS is enabled and you set up the appropriate policies for your tables. You can only get around it by using the **service_role key**, which should always remain hidden.

I still don't know what's to stop someone from using the key to create random users though!

**Comment by zivtamary** [comment]
any elaboration on this from someone experienced in supabase?

**Comment by andrewrjohn** [comment]
The `anon` key is intended to be "exposed" on the client side. In order to fetch data from the DB on the client side without users being logged in, the Supabase client uses the `anon` key.

**Reply by zillBoy** [reply]
I see, thanks @andrewrjohn for the explanation.  
I better make sure that my RSL Policies are properly set up 😄👍. Thanks
